### DISCUSSION TITLE
What does "connections" mean in Supabase terms? How many people can connect to my database?

### DISCUSSION THREAD

**Discussion by mrdbourke** [supabase/supabase#19424] [Questions] [answered]
Hi there,

My brother and I are using Supabase as our backend for an iOS App. 

We're on the Pro tier mentioned here: https://supabase.com/pricing 

Since the app has launched, we're wondering what we should be monitoring for our database going forward.

Our main question is related to "connections".

On the Pro pricing tier, it states that it comes with 500 connections.

<img width="288" alt="Screenshot 2023-12-05 at 12 10 50 pm" src="https://github.com/supabase/supabase/assets/16750345/795894e1-65e5-479a-8b02-5347742e41f3">

Does this mean that 500 people can use the app simultaneously? 

And then after that the database is "full" and more people can't connect?

If that's the case, we'd like to upgrade it so potentially 1000s of people could use the app simultaneously.

Ideally, we'd love our app to have the potential to support 1000s of people writing and reading from the database (rows + images in storage).

Is there a best practice on doing so? 

Thank you, 

Daniel

**Comment by GaryAustin1** [comment]
Realtime connections have nothing to do with database access.  That corresponds to a websocket per user.

If you are using the REST API to access database, then it has its own pooler and there is not real set limit on number of users until you overload the compute of your instance.

**Reply by mrdbourke** [reply]
Hey @GaryAustin1 , 

Thank you for your answer.

Can you expand on the "until you overload the compute of your instance" point?

Is there a recommended/stats breakdown of this somewhere? 

E.g. I'm looking at this table here: 

<img width="1139" alt="Screenshot 2023-12-05 at 1 09 13 pm" src="https://github.com/supabase/supabase/assets/16750345/d5e2bea0-2f58-4a9b-837b-fb59323b91b3">

And from what I'm gathering, the CPU, Memory, Max Disk Bandwidth, Baseline Disk Bandwidth are what I should paying attention to?

Say I have 1000 users all writing 1 row + storing 1 image per second, is this an example of how I could calculate the usage? And then purchase an add-on pre-emptively?

Or should I just continually monitor the database usage and upgrade when necessary?

**Reply by GaryAustin1** [reply, marked as answer]
It is almost impossible to estimate your CPU/memory/disk I/O based on "IF I DO?"  If you do a 1 row select from a 1M row table with complicated RLS it could use a lot of resource.  But the same operation with no RLS on a smaller table could be sub 1msec.   There are benchmarks I've seen of 1000's of inserts a second on Supabase.  Storage does not really involve much CPU/memory/disk I/O resource as it is going thru S3 for data storage and running on its own server.

You'll need to benchmark some loads your app might do and even then if they don't run for significant time your resource charts won't tell you much.  Certainly easier if your app grows over time as you can watch the resource loads.

**Reply by mrdbourke** [reply]
Thank you! That makes more sense. I appreciate the help!
